src/qtism/data/AssessmentTest.php

```diff
@@ -430,17 +433,859 @@ public function isExclusivelyLinear()
     }
 
     /**
-	 * Whether the AssessmentTest as a TimeLimits component bound to it.
-	 *
-	 * @return boolean
-	 */
+     * Whether the AssessmentTest as a TimeLimits component bound to it.
+     *
+     * @return boolean
+     */
     public function hasTimeLimits()
     {
         return $this->getTimeLimits() !== null;
     }
-    
+
     public function __clone()
     {
         $this->setObservers(new SplObjectStorage());
     }
+    
+    /**
+     * Checks if the current session has a parent, and returns it, if any.
+     * 
+     * Gets the list of sections of this AssessmentTest, and checks if any has the AssessmentSection
+     * set as parameter in its components. If some has, we keep the last section found (to take the
+     * closest parent).
+     * 
+     * @param $component AssessmentSection The section from which we want to find the parent.
+     *
+     * @return AssessmentSection|null The parent of the AssessmentSection set as parameter, if any.
+     */
+
+    private function checkRecursion($component)
+    {
+        $sectparent = null;
+
+        foreach ($this->getComponentsByClassName("assessmentSection") as $key => $sect)
+        {
+            if (in_array($component,
+                $sect->getComponentsByClassName("assessmentSection")->getArrayCopy())) {
+                $sectparent = $sect;
+            }
+
+            if ($sect->getIdentifier() == $component->getIdentifier()) {
+                break;
+            }
+        }
+
+        return $sectparent;
+    }
+
+    /**
+     * Returns the first AssessmentItem that will be prompted if a branch targets the
+     * QtiComponent set as parameter.
+     *
+     * This method depends heavily of the QtiClass of the QtiComponent. Some require multiples loops where we need to
+     * find the firstItem from another QtiComponent : this is then done iteratively.
+     * 
+     * @param $component \qtism\data\QtiComponent The QtiComponent targeted by a branch.
+     *
+     * @return \qtism\data\AssessmentItem|null The first AssessmentItem that will be prompted if a branch targets the
+     * QtiComponent set as parameter. Returns null, if there are no more AssessmentItem because the end of the test
+     * has been reached.
+     */
+
+    public function getFirstItem($component)
+    {
+        $current_cmp = $component;
```

Comment:
I know this one will be a pain, but I would like you to respect the variable naming convention we use across the SDK. Instead of having underscore-separated ("_") variable names, we use camel-case notation.

Please use search and replace functionalities as much as you can the absorb this issue, in order to avoid human errors.